code/src/main/java/com/onlinecontacttracing/servlets/messaging/MessageSendingServlet.java

```diff
@@ -1,36 +1,75 @@
 package com.onlinecontacttracing.servlets;
 
-import com.google.api.client.json.jackson2.JacksonFactory;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
 import java.io.IOException;
-import java.util.Iterator;
-import com.google.api.client.googleapis.auth.oauth2.GoogleIdToken;
+import java.util.logging.Logger;
+import com.google.api.client.googleapis.auth.oauth2.GoogleIdTokenVerifier;
 import com.google.api.client.googleapis.auth.oauth2.GoogleIdToken.Payload;
+import com.google.api.client.googleapis.auth.oauth2.GoogleIdToken;
+import com.google.api.client.googleapis.javanet.GoogleNetHttpTransport;
 import com.google.api.client.http.javanet.NetHttpTransport;
-import com.googlecode.objectify.ObjectifyService;
+import com.google.api.client.json.JsonFactory;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.onlinecontacttracing.storage.PositiveUser;
+import com.onlinecontacttracing.messaging.EmailSender;
+import com.onlinecontacttracing.storage.CustomizableMessage;
+import com.onlinecontacttracing.messaging.CompiledMessage;
+import com.onlinecontacttracing.messaging.SystemMessage;
+import com.onlinecontacttracing.messaging.LocalityResource;
 import com.onlinecontacttracing.storage.PersonEmail;
-import com.onlinecontacttracing.storage.NotificationBatch;
-import java.util.Collection;
 import java.util.Collections;
-import com.google.api.client.http.HttpTransport;
-import com.google.api.client.googleapis.auth.oauth2.GoogleIdTokenVerifier;
-import com.google.api.client.googleapis.javanet.GoogleNetHttpTransport;
-import javax.servlet.http.HttpServletResponse;
-import javax.servlet.http.HttpServletRequest;
-import com.google.api.client.json.JsonFactory;
-import java.util.logging.Logger;
-import javax.servlet.annotation.WebServlet;
-import javax.servlet.http.HttpServlet;
 import java.security.GeneralSecurityException;
+import com.onlinecontacttracing.storage.NotificationBatch;
+import static com.googlecode.objectify.ObjectifyService.ofy;
 
-@WebServlet("/message-sender")
-public class MessageSender extends HttpServlet {
 
+@WebServlet("/send-messages")
```

Comment:
nit - this should go directly on the class, the javadoc should be above it.

Reply:
done